webapp/src/sidepanel.tsx

```diff
@@ -187,28 +171,27 @@ export class Sidepanel extends data.Component<SidepanelProps, SidepanelState> {
                         {showFileList && <filelist.FileList parent={parent} />}
                         {showFullscreenButton && <div id="miniSimOverlay" role="button" title={lf("Open in fullscreen")} onClick={this.handleSimOverlayClick} />}
                     </div>
-                    {isTabTutorial && <div className="tutorial-controls">
-                        {backButton}
-                        <Button icon="lightbulb" disabled={true} className="tutorial-hint" />
-                        {nextButton}
-                    </div>}
-                </TabContent>
-                {tutorialOptions && <TabContent name={TUTORIAL_TAB} icon="icon tasks" showBadge={activeTab !== TUTORIAL_TAB} onSelected={this.tryShowTutorialTab} ariaLabel={lf("Open the tutorial tab")}>
-                    <TutorialContainer
-                        parent={parent}
-                        tutorialId={tutorialOptions.tutorial}
-                        name={tutorialOptions.tutorialName}
-                        steps={tutorialOptions.tutorialStepInfo}
-                        currentStep={tutorialOptions.tutorialStep}
-                        tutorialOptions={tutorialOptions}
-                        hideIteration={tutorialOptions.metadata?.hideIteration}
-                        hasTemplate={!!tutorialOptions.templateCode}
-                        preferredEditor={tutorialOptions.metadata?.preferredEditor}
-                        onTutorialStepChange={onTutorialStepChange}
-                        onTutorialComplete={onTutorialComplete}
-                        setParentHeight={this.setComponentHeight} />
-                </TabContent>}
-            </TabPane>
+                </div>
+            </div>
+            {tutorialOptions &&
+                <div id="tutorialWrapper" className={this.props.tutorialSimSidebar ? "topInstructionsWrapper" : ""} style={tutorialHeightStyle}>
+                    <div className={outerTutorialContainerClassName} style={tutorialHeightStyle}>
+                        <TutorialContainer
+                            parent={parent}
+                            tutorialId={tutorialOptions.tutorial}
+                            name={tutorialOptions.tutorialName}
+                            steps={tutorialOptions.tutorialStepInfo}
+                            currentStep={tutorialOptions.tutorialStep}
+                            tutorialOptions={tutorialOptions}
```

Comment:
I know this code hasn't changed from before, but I'm wondering what the benefit is to passing so many props in like this rather than just extracting all this info from `tutorialOptions` from within the component?
Maybe it's just style/preference, but I think it would be better to pass in the minimum that is necessary.

Reply:
@jwunderl can probably say more on this than I can, but I think you're correct that it's mostly just style/preference.

By some metrics, doing it this way is actually more "minimal" than just passing in the tutorialOptions object would be. It's more lines of code, but less actual data than the entire tutorialOptions object. It also means the TutorialContainer is not dependent on the structure of the tutorialOptions object, so it could theoretically be used elsewhere (by something that doesn't have a tutorialOptions object) without the need to refactor. I don't know how realistic that scenario is, but it's one reason you might avoid just passing in the entire object.

Reply:
Oh yea that's a good point. I think the biggest thing that surprised me in this case is that we also pass in the entire `tutorialOptions` object.

Reply:
Yeah that's basically the intention, e.g. this leftover comment with that as the intention https://github.com/microsoft/pxt/blob/stable8.5/webapp/src/components/tutorial/TutorialContainer.tsx#L24

We have a few huge interfaces that end up making refactoring clunky at times (e.g. `IProjectView`) and *ideally* passing just what you need down to the components can make refactoring / component reuse a bit easier. as noted this one is a bit of a stretch as we are unpacking a decent amount of it here, but for what it's worth it is also sometimes nice to have the things the component cares about explicitly listed out.

In theory there are other benefits - e.g. perf improvements if `tutorialOptions` is changing a lot but the props you care about is not as react won't have to call rerender on every change, and avoiding bugs from missed renders if e.g. `tutorialOptions` is mutated in place as the props you care about are viewed explicitly (though that would typically just be considered masking the problem as the mutation would be the true issue), but they're all relatively minor.

Reply:
I hadn't even noticed that we also pass in `tutorialOptions`. I agree that's a bit strange, but I suppose Joey's point about explicitly listing out the requirements still applies to an extent.

Reply:
Yeah, passing in the full object was because it was tedious to unpack completely when a few calls deeply nested in required it (which, I suppose, kinda proves the point on making refactors harder in a slightly roundabout way)